### DISCUSSION TITLE
Micronaut Serialization - how to include empty list properties in response body?

### DISCUSSION THREAD

**Discussion by jehrenzweig-leagueapps** [micronaut-projects/micronaut-core#7289] [Q&A] [answered]
I'm running into a problem regarding serialization of HTTP responses.  Under normal circumstances, I get the following HTTP response body from a Micronaut endpoint where a `List<Stuff>` property contains data:
```
{
    "items": [
        {
            "id": "A",
            "name": "Ayy"
        },
        {
            "id": "B",
            "name": "Bee"
        }
    ]
}
```

However, if the `List<Stuff>` property contains an empty list this is the HTTP response I receive:
```
{}
```

Here is what I would _like_ to receive when the `List<Stuff>` property is an empty list:
```
{
    "items": []
}
```

I tried adding the following code to my `application.yml` file, but it had no effect:
```
jackson:
  serialization-inclusion: always
```

Any suggestions on what else to try and/or how to go about debugging this?  Thanks in advance!

**Comment by yawkat** [comment, marked as answer]
You can inject a JsonMapper into a unit test and use it to serialize your object manually.

**Reply by jehrenzweig-leagueapps** [reply]
I'm not trying to test anything, though -- I want to change the behavior of serialization in the application, so that list properties with empty list values (e.g. `listOf()`) aren't removed by Micronaut's serializer.  That's what the `jackson.serialization-inclusion` config value in `application.yml` is designed to help manage, right?

**Reply by jehrenzweig-leagueapps** [reply]
FWIW I also tried decorating my POJO class with the `@Introspected` annotation, and the POJO list property with the `@JsonProperty` and `@field:JsonProperty` annotations.  However, this had no effect on the serialized object I received in the HTTP response body.

**Reply by yawkat** [reply]
yes it should work, and it does in my tests. so it's something in your code, and you can use JsonMapper to find out what it is.

**Reply by jehrenzweig-leagueapps** [reply]
I switched from using Micronaut Serialization to Jackson Databind and it immediately started working, both using settings in the `application.yml` file or `@JsonInclude` annotations on the POJO properties.  Not sure what I was doing, but I'll stick with Jackson Databind for now.

**Reply by yawkat** [reply]
Yes the `jackson.serialization-inclusion` property will only work when using jackson. The key when using serde is `micronaut.serde.serialization.inclusion`

**Reply by jehrenzweig-leagueapps** [reply]
Ahhh... that's what I was missing.  And now I see that info in the [`Configuration` documentation](https://micronaut-projects.github.io/micronaut-serialization/latest/guide/configurationreference.html#io.micronaut.serde.config.SerializationConfiguration) -- thank you!
